mp2-v1/src/query/batching_planner.rs

```diff
@@ -0,0 +1,458 @@
+            {
+                let sorted_rows = matching_rows.into_iter().collect::<BTreeSet<_>>();
+                stream::iter(sorted_rows.iter())
+                    .then(async |&row_key| {
+                        compute_input_for_row(
+                            &row_cache,
+                            row_key,
+                            index_value,
+                            &index_path,
+                            column_ids,
+                        )
+                        .await
+                    })
+                    .collect::<Vec<RowWithPath>>()
+                    .await
+            } else {
+                //ToDO: find non-existence rows
+                let proven_node = non_existence_inputs
+                    .find_row_node_for_non_existence(index_value)
+                    .await
+                    .expect(
+                        format!("node for non-existence not found for index value {index_value}")
+                            .as_str(),
+                    );
+                let row_input = compute_input_for_row(
+                    non_existence_inputs.row_tree,
+                    &proven_node,
+                    index_value,
+                    &index_path,
+                    column_ids,
+                )
+                .await;
+                vec![row_input]
+            };
+            proven_rows
+        })
+        .concat()
+        .await
+        .chunks(CHUNK_SIZE)
+        .map(|chunk| chunk.to_vec())
+        .collect_vec())
+}
+
+/// Key for nodes of the `UpdateTreeForChunks<NUM_CHUNKS>` employed to
+/// prove chunks of rows.
+/// The key is composed by 2 integers:
+/// - The `level` of the node in the `UpdateTree`, that is the number of
+///   ancestor nodes between the node and the root of the tree
+/// - The `position` of the node in the tree among the nodes with the same
+///   `level`. The position is basically an identifier to uniquely identify
+///   a node among all the nodes in the same level. It is computed recursively
+///   from the position `parent_pos` of the parent node and the number of left
+///   siblings `num_left` of the node as `parent_pos*ARITY + num_left`
+/// For instance, consider the following tree, with arity 3:
+///                 A
+///
+///     B           C           D
+///
+/// E   F   G       H   I    
+/// The nodes in this tree will be identified by the following keys:
+/// ```text
+///                             (0,0)
+///
+///        (1,0)                (1,1)               (1,2)
+///
+/// (2,0)  (2,1)  (2,2)         (2,3)  (2,4)
+/// ```      
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Default)]
+pub struct UpdateTreeKey<const ARITY: usize>((usize, usize));
+
+impl<const ARITY: usize> UpdateTreeKey<ARITY> {
+    /// Compute the key of the child node of `self` that has `num_left_siblings`
+    /// left siblings
+    fn children_key(&self, num_left_siblings: usize) -> Self {
```

Comment:
Not sure I follow: this method should compute only the children keys, according to how [the `position` is defined in an `UpdateTreeKey`](https://github.com/Lagrange-Labs/mapreduce-plonky2/blob/feat/batching_integration_test/mp2-v1/src/query/batching_planner.rs#L164). So, since `num_left_siblings` should be in `[0, ARITY-2]`, since each node will have at most `ARITY` children, the `position` should identify a node unambiguously among all the nodes at the same level of the tree. What am I missing?